flowermd/base/simulation.py

```diff
@@ -1184,6 +1184,7 @@ def save_simulation(self, file_path="simulation.pickle"):
             "log_write_freq": self.log_write_freq,
             "gsd_max_buffer_size": self.maximum_write_buffer_size,
             "seed": self.seed,
+            "constraint": self.constraint,
```

Comment:
Move this down to `sim_dict`, the `sim_kwargs` is just a dict of strings, ints, etc...not the objects that are being wrapped up in the pickle file.